python/lance/pytorch/data.py

```diff
@@ -105,4 +141,16 @@ def __iter__(self):
                 columns=self.columns, batch_size=self.batch_size, filter=self.filter
             )
             for batch in scan.to_reader():
-                yield [to_numpy(arr) for arr in batch.columns]
+                tensors = [to_tensor(arr) for arr in batch.columns]
+                if self.mode == "batch":
+                    if self.transform is not None:
+                        tensors = self.transform(*tensors)
+                    if len(tensors) == 1:
```

Comment:
Do we always want to eliminate a dimension if this is true?

Reply:
Good point. This is for the case where you only read one column (i.e., reading images for inference), to avoid unpack from the caller side.

Maybe I should check `self.columns` instead?